### PR TITLE
[Docs] Insert missing Flow variables

### PR DESCRIPTION
**Pull request by luarss**
Two key changes:
1) Insert missing flow variables 
2) Update platform-specific flow variable aesthetic for ease of viewing. I understand `?=`/ `=` come from the makefile syntax, but might have a big cognitive load on the user.

### CODE REVIEW

File: docs/user/FlowVariables.md [hunk 1]
```diff
@@ -33,9 +33,9 @@ public platforms supported by the OpenROAD flow.
 
 
 Note:
--   = indicates default definition assigned by the tool
--   ?= indicates that the variable value may be reassigned with design `config.mk`
--   N/A indicates that the variable/files is not supported currently.
+-   ☑️ indicates default definition assigned by the tool
+-   ✔️ indicates that the variable value may be reassigned with design `config.mk`
+-   ❌indicates that the variable/files is not supported currently.
```
Comment (vvbandeira) [inline docs/user/FlowVariables.md]: I think we should keep `=` and `?=` as they correlate with the syntax in `.mk` files. Variables that are not supported should not be documented as there's no benefit.

Comment (luarss) [reply docs/user/FlowVariables.md]: Some of the variables are supported for a few/ only one platform - should we keep them?

Comment (vvbandeira) [reply docs/user/FlowVariables.md]: Yes, but in an isolated section. Here we should keep variables that apply to all. We can add them as a subsection under "Optional Variables" at the end of the document.

**Comment by luarss** [comment]
Moved the `N/A` variables. Now there is only `=` and `?=`

File: docs/user/FlowVariables.md [hunk 2]
```diff
@@ -233,11 +229,18 @@ configuration file.
...
-| `CORNER`                 | PVT corner library selection.                                                                      |
+| `CORNER`                 | PVT corner library selection. Only available for ASAP7 and GF180 PDK.                              |
 | `DESIGN_NICKNAME`        | DESIGN_NICKNAME just changes the directory name that ORFS outputs to be DESIGN_NICKNAME instead of DESIGN_NAME in case DESIGN_NAME is unwieldy or conflicts with a difference design.                                                                                    |
 | `ABC_AREA`               | Strategies for Yosys ABC synthesis: Area/Speed. Default ABC_SPEED.                                 |
 | `PWR_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |
 | `GND_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |
+| `BLOCKS`                 | Blocks passed as blackbox into yosys. Do note that you have to specify block-specific inputs file in the directory mentioned by [Makefile](../main/flow/Makefile) |
```
Comment (vijayank88) [inline docs/user/FlowVariables.md]:
```suggestion
| `BLOCKS`                 | Blocks is a hardened macros through ORFS and integrate to top level design. Do note that you have to specify block-specific inputs file in the directory mentioned by [Makefile](../main/flow/Makefile). |

```

File: docs/user/FlowVariables.md [hunk 3]
```diff
@@ -233,11 +229,18 @@ configuration file.
...
+| `CORNER`                 | PVT corner library selection. Only available for ASAP7 and GF180 PDK.                              |
 | `DESIGN_NICKNAME`        | DESIGN_NICKNAME just changes the directory name that ORFS outputs to be DESIGN_NICKNAME instead of DESIGN_NAME in case DESIGN_NAME is unwieldy or conflicts with a difference design.                                                                                    |
 | `ABC_AREA`               | Strategies for Yosys ABC synthesis: Area/Speed. Default ABC_SPEED.                                 |
 | `PWR_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |
 | `GND_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |
+| `BLOCKS`                 | Blocks passed as blackbox into yosys. Do note that you have to specify block-specific inputs file in the directory mentioned by [Makefile](../main/flow/Makefile) |
+| `CDL_FILES`              | Insert additional Circuit Description Language (`.cdl`) netlist files
```
Comment (vijayank88) [inline docs/user/FlowVariables.md]:
```suggestion
| `CDL_FILES`              | Insert additional Circuit Description Language (`.cdl`) netlist files.

```

File: docs/user/FlowVariables.md [hunk 4]
```diff
@@ -233,11 +229,18 @@ configuration file.
...
 | `DESIGN_NICKNAME`        | DESIGN_NICKNAME just changes the directory name that ORFS outputs to be DESIGN_NICKNAME instead of DESIGN_NAME in case DESIGN_NAME is unwieldy or conflicts with a difference design.                                                                                    |
 | `ABC_AREA`               | Strategies for Yosys ABC synthesis: Area/Speed. Default ABC_SPEED.                                 |
 | `PWR_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |
 | `GND_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |
+| `BLOCKS`                 | Blocks passed as blackbox into yosys. Do note that you have to specify block-specific inputs file in the directory mentioned by [Makefile](../main/flow/Makefile) |
+| `CDL_FILES`              | Insert additional Circuit Description Language (`.cdl`) netlist files 
+| `DFF_LIB_FILES`          | Technology mapping liberty files for flip-flops                                                    |
```
Comment (vijayank88) [inline docs/user/FlowVariables.md]:
```suggestion
| `DFF_LIB_FILES`          | Technology mapping liberty files for flip-flops.                                                    |

```

File: docs/user/FlowVariables.md [hunk 5]
```diff
@@ -233,11 +229,18 @@ configuration file.
...
+| `DONT_USE_LIBS`          | Set liberty files as `dont_use`                                                                    |
```
Comment (vijayank88) [inline docs/user/FlowVariables.md]:
```suggestion
| `DONT_USE_LIBS`          | Set liberty files as `dont_use`.                                                                    |

```

File: docs/user/FlowVariables.md [hunk 6]
```diff
@@ -233,11 +229,18 @@ configuration file.
...
+| `PRESERVE_CELLS`         | Mark modules to keep from getting removed in flattening.                                           |
+| `SYNTH_ARGS`             | Optional synthesis variables for yosys.                                                            |
+| `VERILOG_TOP_PARAMS`     | Apply toplevel params (if exist)                                                                   |
```
Comment (vijayank88) [inline docs/user/FlowVariables.md]:
```suggestion
| `VERILOG_TOP_PARAMS`     | Apply toplevel params (if exist).                                                                   |

```

File: docs/user/FlowVariables.md [hunk 7]
```diff
@@ -250,4 +253,31 @@ configuration file.
...
+|--------------------------|----------------------------------------------------------------------------------------------------|
+| `MACRO_WRAPPERS`         | The wrapper file that replace existing macros with their wrapped version.                          |
+
+
+## Clock Tree Synthesis
+
+
+| Variable                 | Description                                                                                        |
+|--------------------------|----------------------------------------------------------------------------------------------------|
+| `CTS_BUF_DISTANCE`       | Distance (in microns) between buffers                                                              |
```
Comment (vijayank88) [inline docs/user/FlowVariables.md]:
```suggestion
| `CTS_BUF_DISTANCE`       | Distance (in microns) between buffers.                                                              |

```

File: docs/user/FlowVariables.md [hunk 8]
```diff
@@ -250,4 +253,31 @@ configuration file.
...
+| `CTS_CLUSTER_DIAMETER`   | Maximum diameter (in microns) of sink cluster. Default 20.                                         |
+| `CTS_CLUSTER_SIZE`       | Maximum number of sinks per cluster. Default 50.
+
+
+## Routing
+| Variable                 | Description                                                                                        |
+|--------------------------|----------------------------------------------------------------------------------------------------|
+| `FASTROUTE_TCL`          | Specifies a Tcl scripts with commands to run before FastRoute                                      |
```
Comment (vijayank88) [inline docs/user/FlowVariables.md]:
```suggestion
| `FASTROUTE_TCL`          | Specifies a Tcl scripts with commands to run before FastRoute.                                      |

```

File: docs/user/FlowVariables.md (hunk 2 above)
Comment (vijayank88) [reply docs/user/FlowVariables.md]: @maliberty is description for `BLOCKS` is correct?

**Review by vijayank88 — CHANGES_REQUESTED** [review]

File: docs/user/FlowVariables.md (hunk 2 above)
Comment (luarss) [reply docs/user/FlowVariables.md]: @maliberty Any thoughts?

Comment (rovinski) [reply docs/user/FlowVariables.md]: Not quite. I would phrase it as this:
`Blocks used as hard macros in a hierarchical flow.`

**Review by vvbandeira — APPROVED** [review]
Good to go after fixing the conflict.

**Comment by luarss** [comment]
@vvbandeira Should be fixed now, just waiting for the CI checks.

**Comment by vvbandeira** [comment]
I see that this PR has an update to `tools/OpenROAD` I don't think this is required for the context of the PR. Please ping me in Slack if you would like help to remove this change from your branch.

**Comment by luarss** [comment]
@vvbandeira Removed the changes to tools/OpenROAD.

**Comment by dralabeing** [comment]
@vvbandeira is this ready to be merged?

**Pull request merged**